serverless/src/index.ts

```diff
@@ -89,110 +32,11 @@ export const handler = async (event: InputEvent, _: any) => {
       };
     }
 
-    // find lambdas then filter out the lamba's that are flagged to be excluded from instrumentation.
-    const lambdas = findLambdas(resources, event.templateParameterValues).filter(
-      (lambda) => !config.exclude?.includes(lambda.key),
-    );
-    log.debug(`Lambda resources found: ${JSON.stringify(lambdas)}`);
-
-    log.debug("Setting environment variables for Lambda function resources");
-    setEnvConfiguration(config, lambdas);
-
-    // Apply layers
-    if (config.addLayers) {
-      log.debug("Applying Layers to Lambda functions...");
-      errors = applyLayers(
-        region,
-        lambdas,
-        config.pythonLayerVersion,
-        config.nodeLayerVersion,
-        config.dotnetLayerVersion,
-        config.javaLayerVersion,
-      );
-      if (errors.length > 0) {
-        return {
-          requestId: event.requestId,
-          status: FAILURE,
-          fragment,
-          errorMessage: errors.join("\n"),
-        };
-      }
-    }
-
-    if (config.addExtension || config.extensionLayerVersion !== undefined) {
-      errors = applyExtensionLayer(region, lambdas, config.extensionLayerVersion);
-      if (errors.length > 0) {
-        return {
-          requestId: event.requestId,
-          status: FAILURE,
-          fragment,
-          errorMessage: errors.join("\n"),
-        };
-      }
-    }
-
-    // Enable tracing
-    const tracingMode = getTracingMode(config);
-    try {
-      log.debug(`Setting tracing mode to ${TracingMode[tracingMode]} for Lambda functions...`);
-      enableTracing(tracingMode, lambdas, resources);
-    } catch (err) {
-      if (err instanceof MissingIamRoleError) {
-        return {
-          requestId: event.requestId,
-          status: FAILURE,
-          fragment,
-          errorMessage: err.message,
-        };
-      }
+    const lambdaOutput = await instrumentLambdas(event, config);
```

Comment:
Moving Lambda-specific code in this file to `lambda/lambda.ts`, which exposes a function `instrumentLambdas()` to be called in `index.ts`.